Trees/BinarySearchTree.cpp

```diff
@@ -0,0 +1,348 @@
+/*
+    Author:
+    Varun Jayathirtha (www.github.com/varunjm)
+
+    Binary Search Tree:
+    A graph data structure where each node has a unidirectional path to at most
+    two node, none of which are itself or its predecessors, i.e the graph is
+    acyclic. The first node is called the root node. Another important feature
+    of this tree is that each node has values lesser than or equal to itself in
+    the left subtree and values greater than or equal to itself in the right
+    subtree.
+*/
+
+#include "BinarySearchTree.hpp"
+#include <iostream>
+#include <climits>
+#include <stack>
+
+using namespace std;
+
+Node * get_node() {
+    Node * newNode = new Node();
+    newNode->left_child = newNode->right_child = nullptr;
+
+    return newNode;
+}
+
+BinarySearchTree :: BinarySearchTree() {
+    root = get_node();
+    root->value = INT_MAX;
```

Comment:
Instead of assigning `INT_MAX` directly, declare a global constant integer named `INF`, set it to `INT_MAX` and use that in place of `INT_MAX`.